aptos-move/block-executor/src/txn_last_input_output.rs

```diff
@@ -242,9 +269,27 @@ impl<T: Transaction, O: TransactionOutput<Txn = T>, E: Debug + Send + Clone>
         )
     }
 
+    pub(crate) fn record_finalized_groups(
+        &self,
+        txn_idx: TxnIndex,
+        finalized_groups: Vec<(T::Key, T::Value, Vec<(T::Tag, Arc<T::Value>)>)>,
+    ) {
+        match &self.outputs[txn_idx as usize]
+            .load_full()
+            .expect("Output must exist")
+            .output_status
+        {
+            ExecutionStatus::Success(t) | ExecutionStatus::SkipRest(t) => {
+                t.record_groups(finalized_groups);
+            },
+            ExecutionStatus::Abort(_) => {},
+        };
+    }
+
     // Called when a transaction is committed to record WriteOps for materialized aggregator values
-    // corresponding to the (deltas) in the recorded final output of the transaction.
-    pub(crate) fn record_delta_writes(
+    // corresponding to the (deltas) in the recorded final output of the transaction, as well as
+    // finalized (resource) group updates (which must be stored internally).
```

Comment:
There is `record_finalized_groups` to update with groups. This `record_additional_writes` function also records the finalized group updates. What's the difference? Comments would be helpful.

Reply:
these wrap the record_groups and incorporate_additional_writes, there should be a comment now, and also I renamed record_additional_writes to incorporate_additional_writes, to be consistent & better distinguish (previously forgot to adjust here).